File: 简单/2129.将标题字母大写/main.go
package main

import (
	"fmt"
	"strings"
)

type Test struct {
	Inputer string
	Wanter  string
	Getter  string
	Passed  bool
}

func main() {
	tests := []Test{
		{
			Inputer: "capiTalIze tHe titLe",
			Wanter:  "Capitalize The Title",
		},
		{
			Inputer: "First leTTeR of EACH Word",
			Wanter:  "First Letter of Each Word",
		},
		{
			Inputer: "i lOve leetcode",
			Wanter:  "i Love Leetcode",
		},
	}
	for idx, test := range tests {
		got := capitalizeTitle(test.Inputer)
		tests[idx].Getter = got
		tests[idx].Passed = got == test.Wanter
	}
	for _, test := range tests {
		fmt.Printf("Input: %s\nOutput: %s\nGet: %s\nPassed: %v\n\n",
			test.Inputer, test.Wanter, test.Getter, test.Passed)
	}
}

func capitalizeTitle(title string) string {
	words := strings.Split(title, " ")
	for i, word := range words {
		words[i] = help(word)
	}
	return strings.Join(words, " ")
}

func help(word string) string {
	if len(word) <= 2 {
		return strings.ToLower(word)
	}
	return strings.ToUpper(word[0:1]) + strings.ToLower(word[1:])
}

File: 简单/2129.将标题字母大写/main.py
class Solution:
    def capitalizeTitle(self, title: str) -> str:
        newWords = []
        for word in title.split(" "):
            newWord = word.lower()
            if len(word) > 2:
                newWords.append(newWord[0].upper() + newWord[1:])
            else:
                newWords.append(word.lower())
        words = " ".join(newWords)
        return words


if __name__ == '__main__':
    s = Solution()
    tests = [
        {
            "input": "capiTalIze tHe titLe",
            "output": "",
            "want": "Capitalize The Title",
            "passed": False,
        },
        {
            "input": "First leTTeR of EACH Word",
            "output": "",
            "want": "First Letter of Each Word",
            "passed": False,
        },
        {
            "input": "i lOve leetcode",
            "output": "",
            "want": "i Love Leetcode",
            "passed": False,
        }
    ]
    for test in tests:
        got = s.capitalizeTitle(test["input"])
        test["output"] = got
        if got == test["want"]:
            test["passed"] = True
    print(tests)


